### PR TITLE
Issue1390 document projection db support

### PR DESCRIPTION
**Pull request by bedaka**

### CODE REVIEW

File: factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/extend-projection-db-support.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+services. Apart from your Db specific implementation you'll need to implement the "FactStreamPositionAware",
+"WriterTokenAware" and "Named" interfaces.
+
+### Projections without Transactional Safety
+
+- Implement the `Projection` interface
+- Override the Getter and Setter for the FactStreamPosition, which represents the information up to which event your projection has consumed the event stream.
+  While the actual implementation will depend on your choice of datastore, one central table per service should be sufficient.
+- Override the `acquireWriteToken` method to provide a locking mechanism for your projection. This is necessary to ensure that only one instance of your projection is processing
+  the event stream and therefore writing to the database at a time.
```
Comment (mamup11) [inline factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/extend-projection-db-support.md:53]:
```suggestion
  the event stream and therefore only one instance is writing to the database at a time.
```

File: factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/extend-projection-db-support.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+### Projections with Transactional Safety
+
+There are two types of transaction aware projections. No matter which one you choose the implementation will mostly the
```
Comment (mamup11) [inline factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/extend-projection-db-support.md:57]:
```suggestion
There are two types of transaction aware projections. No matter which one you choose the implementation will mostly be the
```

File: factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/extend-projection-db-support.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+2. Define a place to store your projection's state (the last processed event). For example this can be a table in your
+   data store that holds one entry for every projection within your service. (FactStreamPositionAware)
+3. Does your datastore provide transactionality? If yes, you can leverage this to batch changes together when updating
+   your projection in order to increase performance and to provide atomicity to your changes. Otherwise, go with a
+   basic projection, but keep in mind that this approach has its limitations in terms of atomicity and throughput.
+
+### General Structure
+
+Projection implementations are usually provided via an abstract class that is extended by the actual projections within your
+services. Apart from to your desired class hierarchy, projections need to implement the `FactStreamPositionAware`,
```
Comment (bedaka) [inline factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/extend-projection-db-support.md:47]:
```suggestion
services. Apart from your desired class hierarchy, projections need to implement the `FactStreamPositionAware`,
```

**Review by bedaka — COMMENTED** [review]
Reads great :)

**Comment by uweschaefer** [comment]
thanks @bedaka and @mamup11 !

**Pull request merged**